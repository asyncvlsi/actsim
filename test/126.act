function testme (int a[3]) : int
{
  chp {
    self := a[0] + a[1] + a[2]
  }
}

defproc test()
{
  int x[4], y[3];

  chp {
    (;i:3: y[i] := i);
    (;i:4: x[i] := 2*i);
    x[0] := testme (x);
    log ("got: ", testme (y))
  }
}
