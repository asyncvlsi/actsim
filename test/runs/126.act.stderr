Typechecking failed on CHP assignment
  stmt: x[0] := testme(x)
	Function `testme<>': arg #0 has an incompatible type
